include/seqan3/range/view/single_pass_input.hpp

```diff
@@ -118,8 +113,15 @@ class single_pass_input_view : public detail::view_base
     ~single_pass_input_view() = default;
 
     //!\brief Construction from the underlying view.
-    single_pass_input_view(urng_t && urng) :
-        view_state_ptr{new view_state{std::forward<urng_t>(urng), seqan3::begin(urng)}}
+    single_pass_input_view(urng_t _urng) :
+        urng{_urng},
+        cached_urng_iter{seqan3::begin(urng)}
+    {}
+
+    //!\brief Construction from InputRange type.
```

Comment:
I am pretty sure construction via view will go through this interface as well, because views are also input ranges and this constructer is more constrained than the above.

Reply:
Well, if the view is a lvalue it will go through this, but in most cases it is a rvalue so it must take the other constructor. 
I though about it and I think that view::all, when passed a view should return the view itself and make it a noop then. But I am not sure, if this is considered by the standard or if this might have other implications, as you might want to enforce the view::all wrapper or some kind of thing.